src/solvers/dg/3d/dg.jl

```diff
@@ -216,41 +219,42 @@ end
-  fstar3_threaded  = A4dp1_z[A4dp1_y(undef, n_variables, n_nodes, n_nodes, n_nodes+1)
-                             for _ in 1:Threads.nthreads()]
-  fstar_upper_left_threaded  = A3d[A3d(undef, n_variables, n_nodes, n_nodes)
-                                   for _ in 1:Threads.nthreads()]
-  fstar_upper_right_threaded = A3d[A3d(undef, n_variables, n_nodes, n_nodes)
-                                   for _ in 1:Threads.nthreads()]
-  fstar_lower_left_threaded  = A3d[A3d(undef, n_variables, n_nodes, n_nodes)
-                                   for _ in 1:Threads.nthreads()]
-  fstar_lower_right_threaded = A3d[A3d(undef, n_variables, n_nodes, n_nodes)
-                                   for _ in 1:Threads.nthreads()]
-  u_large_threaded = A3d[A3d(undef, n_variables, n_nodes, n_nodes)
-                         for _ in 1:Threads.nthreads()]
+  f1_threaded      = A5d[A5d(undef, n_variables, n_nodes, n_nodes, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  f2_threaded      = A5d[A5d(undef, n_variables, n_nodes, n_nodes, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  f3_threaded      = A5d[A5d(undef, n_variables, n_nodes, n_nodes, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  fstar1_threaded  = A4dp1_x[A4dp1_x(undef, n_variables, n_nodes+1, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  fstar2_threaded  = A4dp1_y[A4dp1_y(undef, n_variables, n_nodes, n_nodes+1, n_nodes) for _ in 1:Threads.nthreads()]
+  fstar3_threaded  = A4dp1_z[A4dp1_y(undef, n_variables, n_nodes, n_nodes, n_nodes+1) for _ in 1:Threads.nthreads()]
+  fstar_upper_left_threaded  = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  fstar_upper_right_threaded = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  fstar_lower_left_threaded  = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  fstar_lower_right_threaded = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  fstar_tmp1_threaded        = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  noncons_diamond_upper_left_threaded  = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  noncons_diamond_upper_right_threaded = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  noncons_diamond_lower_left_threaded  = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  noncons_diamond_lower_right_threaded = A3d[A3d(undef, n_variables, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+
+  indicator_threaded  = [zeros(1, n_nodes, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  modal_threaded      = [zeros(1, n_nodes, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  modal_tmp1_threaded = [zeros(1, n_nodes, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
+  modal_tmp2_threaded = [zeros(1, n_nodes, n_nodes, n_nodes) for _ in 1:Threads.nthreads()]
```

Comment:
Why use `zeros` here and `undef` elsewhere? If there is a particular reason, it would be good to leave a comment IMHO.

Reply:
There is no specific reason. I'll change it.